Spec/RealtimeClientPresence.swift

```diff
@@ -2990,6 +2998,93 @@ class RealtimeClientPresence: QuickSpec {
+
+                    let additionalMember = ARTPresenceMessage(
+                        clientId: "two",
+                        action: .enter,
+                        connectionId: connectionId,
+                        id: connectionId + ":0:0"
+                    )
+
+                    // Inject an additional member into the myMember set, then force a suspended state
+                    client.simulateSuspended(beforeSuspension: { done in
+                        channel.presenceMap.localMembers.add(additionalMember)
+                        done()
+                    })
+                    expect(client.connection.state).to(equal(.suspended))
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        channel.once(.attached) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+                            partialDone()
+                        }
+                        // Reconnect
+                        client.connect()
+                        // Await Sync
+                        channel.presenceMap.onceSyncEnds { _ in
+                            partialDone()
+                        }
+                    }
+
+                    expect(channel.presenceMap.syncComplete).toEventually(beTrue(), timeout: testTimeout)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        channel.presence.subscribe(.enter) { presence in
+                            expect(presence.clientId).to(equal("two"))
+                            channel.presence.unsubscribe()
+                            partialDone()
+                        }
+                        if channel.presenceMap.syncComplete {
+                            channel.sync { error in
+                                expect(error).to(beNil())
+                                channel.presenceMap.onceSyncEnds { _ in
+                                    partialDone()
+                                }
+                            }
+                        }
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.presence.get { presences, error in
+                            expect(error).to(beNil())
+                            guard let presences = presences else {
+                                fail("Presences is nil"); done(); return
+                            }
+                            expect(channel.presenceMap.syncComplete).to(beTrue())
+                            expect(presences).to(haveCount(2))
+                            expect(presences.map({$0.clientId})).to(contain(["one", "two"]))
+                            done()
+                        }
+                    }
+                }
+
```

Comment:
I can't see the bit of this test that tests synthesized leave's effect on the own-members set, as the test name implies? (If it is there than it's odd that it passed, since https://github.com/ably/ably-cocoa/pull/835/files#r278056029 means that the logic wasn't actually taking effect)

Reply:
@SimonWoolf You're right, that part is missing. What's the best way to test the synthesized leave's effect? I'm thinking in forcing a synthesized leave of one existing presence and force a sync. Then after the sync, the user should be updated since the user is re-entered and the final members map should remain the same, no leave has taken effect.

Reply:
A proper end-to-end test that forces realtime to do a synthesized leave is a bit annoying if only because it means the test will take at least 15 seconds. But if you want to do that, something like: enter, force a disconnect, wait 20s, reconnect, wait to see the synthesized leave (& sync showing the member not present), wait for lib to automatically re-enter.

Reply:
@SimonWoolf Thanks, updated the test case 46dd5e7.